core/node/da_clients/src/eigenv2m0/client.rs

```diff
@@ -0,0 +1,135 @@
+    rust_eigenda_signers::signers::private_key::Signer,
+    utils::SecretUrl,
+};
+use rust_eigenda_v2_common::{Payload, PayloadForm};
+use subxt_signer::ExposeSecret;
+use url::Url;
+use zksync_config::{
+    configs::da_client::eigenv2m0::{EigenSecretsV2M0, PolynomialForm},
+    EigenConfigV2M0,
+};
+use zksync_da_client::{
+    types::{ClientType, DAError, DispatchResponse, FinalityResponse, InclusionData},
+    DataAvailabilityClient,
+};
+
+use crate::utils::{to_non_retriable_da_error, to_retriable_da_error};
+
+// We can't implement DataAvailabilityClient for an outside struct, so it is needed to defined this intermediate struct
+#[derive(Debug, Clone)]
+pub struct EigenDAClientV2M0 {
+    client: PayloadDisperser,
+}
+
+impl EigenDAClientV2M0 {
+    pub async fn new(config: EigenConfigV2M0, secrets: EigenSecretsV2M0) -> anyhow::Result<Self> {
+        let url = Url::from_str(
+            config
+                .eigenda_eth_rpc
+                .ok_or(anyhow::anyhow!("Eigenda eth rpc url is not set"))?
+                .expose_str(),
+        )
+        .map_err(|_| anyhow::anyhow!("Invalid eth rpc url"))?;
+        let payload_form = match config.polynomial_form {
+            PolynomialForm::Coeff => PayloadForm::Coeff,
+            PolynomialForm::Eval => PayloadForm::Eval,
+        };
+
+        let payload_disperser_config = PayloadDisperserConfig {
+            polynomial_form: payload_form,
+            blob_version: config.blob_version,
+            cert_verifier_address: config.cert_verifier_addr,
+            eth_rpc_url: SecretUrl::new(url),
+            disperser_rpc: config.disperser_rpc,
+            use_secure_grpc_flag: config.authenticated,
+        };
+        let private_key = secrets
+            .private_key
+            .0
+            .expose_secret()
+            .parse()
+            .map_err(|e| anyhow::anyhow!("Failed to parse private key: {}", e))?;
+        let signer = Signer::new(private_key);
+        let client = PayloadDisperser::new(payload_disperser_config, signer)
+            .await
+            .map_err(|e| anyhow::anyhow!("Eigen client Error: {:?}", e))?;
+        Ok(Self { client })
+    }
+}
+
+#[async_trait::async_trait]
+impl DataAvailabilityClient for EigenDAClientV2M0 {
+    async fn dispatch_blob(
+        &self,
+        _: u32, // batch number
+        data: Vec<u8>,
+    ) -> Result<DispatchResponse, DAError> {
+        let payload = Payload::new(data);
+        let blob_key = self
+            .client
+            .send_payload(payload)
+            .await
+            .map_err(to_retriable_da_error)?;
+
+        Ok(DispatchResponse::from(blob_key.to_hex()))
+    }
+
+    async fn ensure_finality(
+        &self,
+        dispatch_request_id: String,
+    ) -> Result<Option<FinalityResponse>, DAError> {
+        // TODO: return a quick confirmation in `dispatch_blob` and await here
+        Ok(Some(FinalityResponse {
+            blob_id: dispatch_request_id,
+        }))
```

Comment:
seems like the implementation is not finished?
@dimazhornyk I forget if you prefer to wait for the full cert (~10sec) in `dispatch_blob` and make `ensure_finality` a noop for v2, or we can have `ensure_finality` actually do the 10s polling.

Reply:
Also @dimazhornyk https://docs.zksync.io/zk-stack/running/validium needs an update, it doesn't contain the `ensute_finality` function